webhooks/virtualmachine/validation/virtualmachine_validator.go

```diff
@@ -153,6 +154,37 @@ func (v validator) ValidateDelete(*pkgctx.WebhookRequestContext) admission.Respo
 	return admission.Allowed("")
 }
 
+// Updates to VM's image are only allowed if it is a Registered VM (used for failover
+// in disaster recovery).
+func (v validator) validateImageOnUpdate(ctx *pkgctx.WebhookRequestContext, vm, oldVM *vmopv1.VirtualMachine) field.ErrorList {
+	var allErrs field.ErrorList
+
+	if f := pkgcfg.FromContext(ctx).Features; f.VMIncrementalRestore {
+		if annotations.HasRegisterVM(vm) {
+			if !equality.Semantic.DeepEqual(oldVM.Spec.Image, vm.Spec.Image) {
```

Comment:
The image ref is simple, so maybe just add a helper func to compare two of them instead of using a DeepEqual?